### PR TITLE
2.0.0-dev.1.4

### PR DESCRIPTION
**Pull request by lexaknyazev**
* Refined pointers for `DUPLICATE_ELEMENTS`, `UNSUPPORTED_EXTENSION`, and `UNUSED_EXTENSION_REQUIRED`.
* Add new validation check `UNRESERVED_EXTENSION_PREFIX`.
* Small tests cleanup.

### CODE REVIEW

**Comment by lexaknyazev** [comment]
* Added `MATERIAL_ALPHA_CUTOFF_INVALID_MODE`. See https://github.com/KhronosGroup/glTF/issues/1158.

**Comment by lexaknyazev** [comment]
@donmccurdy This PR should fix #49.

**Comment by emackey** [comment]
Sorry for the delay @lexaknyazev... Short week this week, a little swamped here.  Will get to this soon as I can.

**Comment by donmccurdy** [comment]
Alpha cutoff and Firefox/Edge fixes look good to me thanks!

File: ISSUES.md [hunk 1]
```diff
@@ -39,21 +39,23 @@
...
+|22|MESH_PRIMITIVE_TANGENT_POINTS|TANGENT attribute defined for POINTS rendering mode.|Warning|
+|23|MESH_INVALID_WEIGHTS_COUNT|The length of weights array (`%1`) does not match the number of morph targets (`%2`).|Error|
+|24|NODE_MATRIX_TRS|A node can have either a matrix or any combination of translation/rotation/scale (TRS) properties.|Error|
+|25|NODE_MATRIX_DEFAULT|Do not specify default transform matrix.|Information|
+|26|NODE_MATRIX_NON_TRS|Matrix must be decomposable to TRS.|Error|
+|27|NODE_ROTATION_NON_UNIT|Rotation quaternion must be normalized.|Error|
+|28|UNUSED_EXTENSION_REQUIRED|Unused extension '`%1`' cannot be required.|Error|
+|29|UNRESERVED_EXTENSION_PREFIX|Extension uses unreserved extension prefix '`%1`'.|Warning|
+|30|NODE_EMPTY|Empty node encountered.|Information|
+|31|NON_RELATIVE_URI|Non-relative URI found: `%1`.|Warning|
```
Comment (emackey) [inline ISSUES.md]: Do these error numbers do any good for users, given they can change between versions?  For example `NON_RELATIVE_URI` changed from `29` to `31` here.  If these aren't stable, any reason not to get rid of them?

File: lib/src/ext/extensions.dart [hunk 2]
```diff
@@ -68,3 +61,16 @@ final List<Extension> defaultExtensions = <Extension>[
   new CesiumRtcExtension(),
   new Web3dQuantizedAttributesExtension()
 ];
+
+// https://github.com/KhronosGroup/glTF/blob/master/extensions/Prefixes.md
+const List<String> kReservedPrefixes = const <String>[
+  'KHR_',
```
Comment (emackey) [inline lib/src/ext/extensions.dart]: I wonder if we'll need a reminder to update this list from time to time.

File: web/index.html [hunk 3]
```diff
@@ -41,10 +41,11 @@
 </header>
 
 <div id="dropZone">
-    Drop glTF or GLB here.
+    <input id="input" type="file" style="display: none" multiple/>
+    <a id="inputLink">Select</a> glTF or GLB asset or drop it here.
 </div>
-
 <div id="outputZone">
+    <p><em>Validation is performed locally in your browser. Submitted assets are not uploaded.</em></p>
```
Comment (emackey) [inline web/index.html]: 👍

File: ISSUES.md (hunk 1 above)
Comment (lexaknyazev) [reply ISSUES.md]: I've used them back in a day to keep track on the total amount of recognized issues. They serve no purpose now, so let's remove them.

Comment (lexaknyazev) [reply ISSUES.md]: Also, should we sort messages by `code`? Current order is semi-random.

File: lib/src/ext/extensions.dart (hunk 2 above)
Comment (lexaknyazev) [reply lib/src/ext/extensions.dart]: What would you suggest? Adding a comment here or adding a note to `Prefixes.md`, or both?

Comment (emackey) [reply lib/src/ext/extensions.dart]: We've already got the link here in a comment.  I think `Prefixes.md` should get a note (after this is merged) that any changes there should be copied here (or bump the glTF-Validator maintainers to do it).

**Comment by emackey** [comment]
Published `2.0.0-dev.1.4` to npm.

**Comment by donmccurdy** [comment]
https://gltf-viewer.donmccurdy.com/ validation working in Firefox now, thanks!

**Pull request merged**